### PR TITLE
WT-8493 Fast-truncate fails to detect conflicts with prior removes

### PR DESCRIPTION
**Pull request by sauclovian-wt**
Fix behavior of fast-truncate when conflicting with a prior remove.

(check both start times and stop times, not just start times)

Add new test truncate06 that checks this and a variety of related scenarios.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Review by quchenhao — APPROVED** [review]
lgtm

File: test/suite/test_truncate06.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+# test_truncate06.py
+#
+# Test various scenarios with timestamps and truncation.
```
Comment (quchenhao) [inline test/suite/test_truncate06.py:35]: May be clearly describe the scenario of this test.

Comment (sauclovian-wt) [reply test/suite/test_truncate06.py]: Good plan. Fixed.

**Pull request merged**